allennlp/modules/seq2vec_encoders/pytorch_seq2vec_wrapper.py

```diff
@@ -0,0 +1,66 @@
+import torch
+from torch.nn.utils.rnn import pack_padded_sequence
+
+from allennlp.common.tensor import sort_batch_by_length
+from allennlp.modules import Seq2VecEncoder
+from allennlp.common.checks import ConfigurationError
+
+
+class PytorchSeq2VecWrapper(Seq2VecEncoder):
+    """
+    Pytorch's RNNs have two outputs: the hidden state for every time step, and the hidden state at
+    the last time step for every layer.  We just want the second one as a single output.  This
+    wrapper pulls out that output, and adds a :func:`get_output_dim` method, which is useful if you
+    want to, e.g., define a linear + softmax layer on top of this to get some distribution over a
+    set of labels.  The linear layer needs to know its input dimension before it is called, and you
+    can get that from ``get_output_dim``.
+
+    Also, there are lots of ways you could imagine going from an RNN hidden state at every
+    timestep to a single vector - you could take the last vector at all layers in the stack, do
+    some kind of pooling, take the last vector of the top layer in a stack, or many other  options.
+    We just take the final hidden state vector.  TODO(mattg): allow for other ways of wrapping
+    RNNs.
+    """
+    def __init__(self, module: torch.nn.modules.RNNBase) -> None:
+        super(PytorchSeq2VecWrapper, self).__init__()
+        self._module = module
+        if not self._module.batch_first:
+            raise ConfigurationError("Our encoder semantics assumes batch is always first!")
+
+    def get_output_dim(self) -> int:
+        return self._module.hidden_size * (2 if self._module.bidirectional else 1)
+
+    def forward(self,  # pylint: disable=arguments-differ
+                inputs: torch.Tensor,
+                sequence_lengths: torch.LongTensor = None) -> torch.Tensor:
+
+        if sequence_lengths is None:
+            return self._module(inputs)[0][:, -1, :]
```

Comment:
Yeah sorry, this could have been a lot clearer - this is actually important, because they are _not_ the same _if_ you have variable length sequences. By using the state, pytorch collects the correct final state for each sequence, but if you just take the last element from `lstm(inputs)[0]`, this will be zeros for elements which are not as long as the batch. However, if you aren't passing sequence lengths in, we're assuming that your sequences are full, in which case they are the same (and it's clearly a simpler operation, which was why I preferred it here).